### PR TITLE
 Improve monitoring and feedback for errors during workspace cluster selection

### LINKED ISSUE #7829: Improve monitoring and feedback for errors during workspace cluster selection
### Bug description

On start, some workspaces get stuck on "[Allocating resources (from Discord)](https://discord.com/channels/816244985187008514/839379835662368768/935668444924350475)".

[Another example from Twitter.](https://twitter.com/jahilliar/status/1486108607103991812?s=21)

### Steps to reproduce

Unconfirmed, but, try to start a workspace when prebuild load is very high.

### Workspace affected

vipertools-css-eamifbfe80m

### Expected behavior

It should start, instead of getting stuck.

### Example repository

n/a

### Anything else?

The issue occurred for some users on January 25 @ 10:52pm UTC.

### PR DESCRIPTION
**Pull request by geropl**
## Description
This PR:
 - re-tries cluster-selection for a particular workspace instance for up to 2 times, with a 2s interval
 - introduces two new metrics: 'gitpod_server_instance_starts_success_total{retries}' and 'gitpod_server_instance_starts_failed_total'
 - adds a Graph for it on "Meta Overview" [sic]
 - adds an alert `InstanceStartFailures` (incl. [runbook](https://github.com/gitpod-io/runbooks/pull/327))
image

## Related Issue(s)

Fixes #7829 
Runbook PR: https://github.com/gitpod-io/runbooks/pull/327

## How to test
 - start a workspace on [this branch](https://gitpod.io/#https://github.com/gitpod-io/gitpod/pull/8486)
 - `kubectl port-forward deployment grafana 3000`
 - Open that port and navigate to "Team WebApp / Meta Overview"

## Release Notes

```release-note
improve robustness of startWorkspace
improve feedback for errors during cluster selection
improve monitoring for cluster selection errors
```

## Documentation

- [ ] /werft with-helm
- [ ] /werft with-observability
- [x] /werft no-preview

### CODE REVIEW

**Comment by geropl** [comment]
**Note**: the build is read because of some bug with the "observability" deployment process, but **Gitpod and Grafana do work.**
We'll have to override the PR check once this is approved.

File: components/server/src/workspace/workspace-starter.ts [hunk 1]
```diff
@@ -259,6 +242,42 @@ export class WorkspaceStarter {
...
+                }
+
+                // start that thing
+                log.info({ instanceId: instance.id }, 'starting instance');
+                return (await manager.startWorkspace(ctx, startRequest)).toObject();
+            } catch (err: any) {
+                if ('code' in err && err.code !== grpc.status.OK && lastInstallation !== "") {
+                    log.error({ instanceId: instance.id }, "cannot start workspace on cluster, might retry", err, { cluster: lastInstallation });
+                } else {
+                    throw err;
```
Comment (AlexTugarev) [inline components/server/src/workspace/workspace-starter.ts:273]: this still may throw. should it be handled in `actuallyStartWorkspace`?

Comment (geropl) [reply components/server/src/workspace/workspace-starter.ts:273]: My intention was to not touch the error behavior here for now, as it is rather complex, and we rely on this further up the call chain. This PR is only intended to fetch the scheduling issues for now.

File: components/server/src/workspace/workspace-starter.ts [hunk 2]
```diff
@@ -180,45 +184,24 @@ export class WorkspaceStarter {
...
-                        log.error({ instanceId: instance.id }, "cannot start workspace on cluster, might retry", err, { cluster: lastInstallation });
-                    } else {
-                        throw err;
-                    }
                 }
+                await new Promise((resolve) => setTimeout(resolve, INSTANCE_START_RETRY_INTERVAL_SECONDS * 1000));
             }
+
             if (!resp) {
+                increaseFailedInstanceStartCounter("clusterSelectionFailed");
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/server/src/workspace/workspace-starter.ts:201]: Thx, excellent point! :100:  Will add

**Comment by geropl** [comment]
/werft run

Hope to get a green build now with changed flags...

**Review by AlexTugarev — APPROVED** [review]

**Pull request merged**